bindings/rust/bench/src/rustls.rs

```diff
@@ -82,23 +87,38 @@ impl TlsBenchHarness for RustlsHarness {
             ECGroup::X25519 => &X25519,
         };
 
-        let client_config = Arc::new(
-            ClientConfig::builder()
-                .with_cipher_suites(&[cipher_suite])
-                .with_kx_groups(&[kx_group])
-                .with_protocol_versions(&[&TLS13])?
-                .with_root_certificates(Self::get_root_cert_store()?)
-                .with_no_client_auth(),
-        );
-
-        let server_config = Arc::new(
-            ServerConfig::builder()
-                .with_cipher_suites(&[cipher_suite])
-                .with_kx_groups(&[kx_group])
-                .with_protocol_versions(&[&TLS13])?
-                .with_no_client_auth()
-                .with_single_cert(Self::get_cert_chain()?, Self::get_server_key()?)?,
-        );
+        let client_builder = ClientConfig::builder()
+            .with_cipher_suites(&[cipher_suite])
+            .with_kx_groups(&[kx_group])
+            .with_protocol_versions(&[&TLS13])?
+            .with_root_certificates(Self::get_root_cert_store()?);
+
+        let server_builder = ServerConfig::builder()
+            .with_cipher_suites(&[cipher_suite])
+            .with_kx_groups(&[kx_group])
+            .with_protocol_versions(&[&TLS13])?;
+
+        let (client_builder, server_builder) = match handshake_type {
```

Comment:
Why not just use an if statement here?

Reply:
I used a match so it's more clear one branch is for mTLS and one branch isn't. A match is also more clear for me since `client_builder` and `server_builder` are being reassigned because the builder methods consume the builder and return a new one. I know you could do reassignment with an if statement too, but a match feels more natural. However, I'm not super convinced to do it one way or another.